godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Vector2.kt

```diff
@@ -0,0 +1,311 @@
+@file:Suppress("unused", "MemberVisibilityCanBePrivate")
+
+package godot.core
+
+import godot.gdnative.*
+import kotlinx.cinterop.*
+import kotlin.math.*
+
+class Vector2(var x: RealT, var y: RealT) : Comparable<Vector2>, CoreType {
+    //CONSTANTS
+    companion object {
+        const val AXIS_X: Int = 0
+        const val AXIS_Y: Int = 1
+        val ZERO = Vector2(0, 0)
```

Comment:
A problem here is that the user can accidentally mutate the instance of `ZERO` which is shared everywhere. A fix I had in my binding is to do:

```kotlin
val ZERO: Vector2
	get() = Vector2(0, 0)
```

This means that ZERO, will always return a unique instance.

Reply:
Fair point.